atlasdb-remoting-api/src/main/java/com/palantir/atlasdb/config/RemotingClientConfig.java

```diff
@@ -28,6 +28,10 @@
 @Value.Immutable
 public interface RemotingClientConfig {
     RemotingClientConfig DEFAULT = ImmutableRemotingClientConfig.builder().build();
```

Comment:
default is poorly named, we should be explicit about what that means, somewhat unrelated to this PR but was confused when I was seeing it being used

Reply:
I'll rename default to `ALWAYS_USE_LEGACY` (which is the current behaviour).

Reply:
(and, goes without saying, but add explicit specifications in case the real default changes!)